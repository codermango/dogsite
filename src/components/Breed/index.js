import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.css';

class Breed extends Component {

  static propsType = {
    name: PropTypes.string.isRequired,
    image: PropTypes.string,
    subBreeds: PropTypes.shape([{
      name: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    }]),
    text: PropTypes.string.isRequired
  }

  renderContent = () => {
    const { name, image, subBreeds } = this.props.data;

    return subBreeds ? (
      <div className={styles.subBreeds}>
        {subBreeds.map((subBreed) => (
          <div key={subBreed.name} className={styles.subBreed}>
            <h2 className={styles.subBreedName}>{subBreed.name}</h2>
            <img className={styles.image} alt={subBreed.name} src={subBreed.image} />
          </div>
        ))}
      </div>
    ) : (
      <img className={styles.image} alt={name} src={image} />
    );
  }

  render() {
    const { name, text } = this.props.data;

    return (
      <div className={styles.container}>
        <h1 className={styles.breedName}>{name}</h1>
        {this.renderContent()}
        <p>{text}</p>
      </div>
    )
  }
}

export default Breed;
