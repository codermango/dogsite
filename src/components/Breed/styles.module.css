.container {
  border: dotted 1px black;
  margin: 20px;
  padding: 20px 0;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.breedName {
  font-size: 24px;
}

.subBreedName {
  font-size: 16px;
}

.subBreeds {
  display: flex;
  width: 100%;
  justify-content: space-around;
  flex-wrap: wrap;
}

.subBreed {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.image {
  height: 200px;
}