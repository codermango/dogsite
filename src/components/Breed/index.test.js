import React from 'react';
import renderer from 'react-test-renderer';

import Breed from './index.js';

describe('Breed', () => {
  it('Breed renders correctly when there is no subbreeds', () => {
    const props = {
      name: 'testName',
      image: 'testImage',
      text: 'testText'
    };
    const component = renderer.create(<Breed data={props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Breed renders correctly when there is subbreeds', () => {
    const props = {
      name: 'testName',
      subBreeds: [
        { name: 'testSubName1', image: 'testSubImage1' },
        { name: 'testSubName2', image: 'testSubImage2' }
      ],
      text: 'testText'
    };
    const component = renderer.create(<Breed data={props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

