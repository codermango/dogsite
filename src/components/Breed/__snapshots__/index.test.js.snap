// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breed Breed renders correctly when there is no subbreeds 1`] = `
<div
  className="container"
>
  <h1
    className="breedName"
  >
    testName
  </h1>
  <img
    alt="testName"
    className="image"
    src="testImage"
  />
  <p>
    testText
  </p>
</div>
`;

exports[`Breed Breed renders correctly when there is subbreeds 1`] = `
<div
  className="container"
>
  <h1
    className="breedName"
  >
    testName
  </h1>
  <div
    className="subBreeds"
  >
    <div
      className="subBreed"
    >
      <h2
        className="subBreedName"
      >
        testSubName1
      </h2>
      <img
        alt="testSubName1"
        className="image"
        src="testSubImage1"
      />
    </div>
    <div
      className="subBreed"
    >
      <h2
        className="subBreedName"
      >
        testSubName2
      </h2>
      <img
        alt="testSubName2"
        className="image"
        src="testSubImage2"
      />
    </div>
  </div>
  <p>
    testText
  </p>
</div>
`;
